src/AspNet.Security.OAuth.Douyin/DouyinAuthenticationHandler.cs

```diff
@@ -0,0 +1,235 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Net;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace AspNet.Security.OAuth.Douyin;
+
+/// <summary>
+/// Defines a handler for authentication using Douyin.
+/// </summary>
+public partial class DouyinAuthenticationHandler : OAuthHandler<DouyinAuthenticationOptions>
+{
+    public DouyinAuthenticationHandler(
+        [NotNull] IOptionsMonitor<DouyinAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder)
+        : base(options, logger, encoder)
+    {
+    }
+
+    private const string AuthCode = "auth_code";
```

Comment:
Looks like this could just be made local to `TryStandardizeRemoteAuthenticateQuery()`.